components/archiving/archiveDao.js

```diff
@@ -0,0 +1,467 @@
+        const archiveData = {
+          archive_entity_id: data.id,
+          workflow_run_ids_for_offsite_archive:
+            kase.workflowRunIdsForOffsiteArchive,
+          workflow_run_ids_for_vidarr_archival:
+            kase.workflowRunIdsForVidarrArchival,
+          metadata: kase.metadata,
+          archive_target: kase.archiveTarget,
+          archive_with: kase.archiveWith,
+        };
+
+        const archiveQuery = pgp.helpers.insert(
+          archiveData,
+          archiveColsCreate,
+          'archive'
+        );
+        await tx.none(archiveQuery);
+        return;
+      });
+    })
+      .then(() => {
+        resolve();
+      })
+      .catch((err) => {
+        reject(new Error(err));
+      });
+  });
+};
+
+const archiveEntityArchiveDataQueryWithoutUnloadFiles = `SELECT c.${entityIdentifierColumn}, c.${requisitionId}, c.${limsIds}, c.${archiveType}, a.${created}, a.${modified}, a.${wfrIdsForOffsite}, a.${filesCopiedToOffsiteStagingDir}, a.${commvaultJobId}, a.${wfrIdsForVidarrArchival}, a.${filesLoadedIntoVidarrArchival}, a.${filesUnloaded}, a.${metadata}, a.${archiveTarget}, a.${archiveWith}, a.${batchId}, a.${stopProcessing} FROM archive_entity c JOIN archive a ON c.${id} = a.${archiveEntityId}`;
+const archiveEntityArchiveDataQueryWithUnloadFiles = `SELECT c.${entityIdentifierColumn}, c.${requisitionId}, c.${limsIds}, c.${archiveType}, a.${created}, a.${modified}, a.${wfrIdsForOffsite}, a.${unloadFileForOffsite}, a.${filesCopiedToOffsiteStagingDir}, a.${commvaultJobId}, a.${wfrIdsForVidarrArchival}, a.${unloadFileForVidarrArchival}, a.${filesLoadedIntoVidarrArchival}, a.${filesUnloaded}, a.${metadata}, a.${archiveTarget}, a.${archiveWith}, a.${batchId}, a.${stopProcessing} FROM archive_entity c JOIN archive a ON c.${id} = a.${archiveEntityId}`;
+
+const getArchiveQuery = (includeUnloadFiles = false, entityType) => {
+  let query;
+  if (includeUnloadFiles) {
+    query = archiveEntityArchiveDataQueryWithUnloadFiles;
+  } else {
+    query = archiveEntityArchiveDataQueryWithoutUnloadFiles;
+  }
+  query += ` WHERE ${entityIdentifierColumn} = $1 AND ${archiveType} = '${entityType}'`;
+  return query;
+};
+
+const getByArchiveEntityIdentifier = (
+  archiveEntityIdentifier,
+  includeUnloadFiles = false,
+  entityType
+) => {
+  const query = getArchiveQuery(includeUnloadFiles, entityType);
+  return new Promise((resolve, reject) => {
+    db.manyOrNone(query, archiveEntityIdentifier)
+      .then((data) => {
+        resolve(data);
+      })
+      .catch((err) => {
+        if (err.code === qrec.noData) {
+          // no data found
+          resolve([]);
+        } else {
+          reject(new Error(err));
+        }
+      });
+  });
+};
+
+/** Return data or NotFoundError */
+const getByFilesNotCopiedToOffsiteStagingDir = (entityType) => {
+  const query =
+    archiveEntityArchiveDataQueryWithoutUnloadFiles +
+    ` WHERE ${filesCopiedToOffsiteStagingDir} IS NULL AND ${archiveType} = '${entityType}'`;
```

Comment:
Is there a particular reason to use the bare `entityType` here (and in functions below), rather than the SQL injection-proof `${archiveType} = $2` and pass entityType as a parameter to the db call?